main/src/ca/uwaterloo/flix/language/phase/njvm/classes/RecordExtend.scala

```diff
@@ -0,0 +1,142 @@
+  //there each of them holds the capability to call the corresponding method
+  val constructor: Method3[JvmType.String.type, T, MnemonicsType.RecordInterface.type, JvmType.Void.type] = genConstructor
+
+  val getRecordWithField: Method1[JvmType.String.type, MnemonicsType.RecordInterface.type] = genGetRecordWithFieldMethod
+
+  val restrictField: Method1[JvmType.String.type, MnemonicsType.RecordInterface.type] = genRestrictFieldInterfaceMethod
+
+  val _toString: Method0[JvmType.String.type] = genToStringMethod
+
+  val _hashCode: Method0[JvmType.PrimInt.type] = genHashCodeMethod
+
+  val equals: Method1[JvmType.Object.type, JvmType.PrimBool.type] = genEqualsMethod
+
+  private def genConstructor: Method3[JvmType.String.type, T, MnemonicsType.RecordInterface.type, JvmType.Void.type] = {
+
+    cg.mkMethod3(List(Public), "<init>",
+      sig =>
+        sig.getArg0.LOAD[StackNil] |>>
+          JavaRuntimeFunctions.ObjectConstructor.INVOKE |>>
+          sig.getArg0.LOAD |>>
+          sig.getArg1.LOAD |>>
+          field0.PUT_FIELD |>>
+
+          sig.getArg0.LOAD |>>
+          sig.getArg2.LOAD |>>
+          field1.PUT_FIELD |>>
+
+          sig.getArg0.LOAD |>>
+          sig.getArg3.LOAD |>>
+          field2.PUT_FIELD |>>
+          RETURN)
+  }
+
+  private def genGetRecordWithFieldMethod: Method1[JvmType.String.type, MnemonicsType.RecordInterface.type] = {
+
+//    val ret = new Label
+//    val falseCase = new Label
+//
+//    cg.mkMethod1(List(Public, Final), "getRecordWithField",
+//      sig =>
+//        sig.getArg0.LOAD[StackNil] |>>
+//          field0.GET_FIELD |>>
+//          sig.getArg1.LOAD |>>
+//          JavaRuntimeFunctions.StringEquals.INVOKE |>>
+//          IFEQ(falseCase) |>>
+//          sig.getArg0.LOAD |>>
+//          GOTO(ret)|>>
+//          EMIT(falseCase) |>>
+//          EMIT(ret) |>>
+//          RETURN[MnemonicsType.RecordInterface.type]
+//
+//    )
+    cg.mkMethod1(List(Public, Final), "getRecordWithField",
+      _ =>
+        newUnsupportedOperationExceptionInstructions("getRecordWithField shouldn't be called")
+    )
+
+  }
+  private def genRestrictFieldInterfaceMethod: Method1[JvmType.String.type, MnemonicsType.RecordInterface.type] =
+    cg.mkMethod1(List(Public, Final), "restrictField",
+      _ =>
+        newUnsupportedOperationExceptionInstructions("restrictField shouldn't be called")
+    )
+
+  /**
+    * Method which generates the `toString()` method which will always throws an exception, since `toString` should not be called.
+    * Despite in order to stay in line with our format we still return the capability to call the method
+    * The `toString` method is always the following:
+    *
+    * public string toString() throws Exception {
+    * throw new Exception("toString method shouldn't be called");
+    * }
+    */
+  private def genToStringMethod: Method0[JvmType.String.type] =
+    cg.mkMethod0(List(Public, Final), "toString",
```

Comment:
Lets get rid of the List(Public, Final) since it will be used almost everywhere, so it should just be some default.